app/js/enhancedDate.js

```diff
@@ -1,5 +1,72 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function() { 
+    'use strict';
+    var myDate = null;
+    var dateSet = false;     
+
+    return {    
+        getDate: function(returnObj) {  
+            if (!dateSet)  {
+                this.setDate();
+            }
+            if (returnObj) {
+                return myDate;
+            } else {
+                return myDate.getTime();
+            }
+        },
+        setDate: function(newDate) {
+            if (isNaN(Date.parse(newDate))) {
+                myDate = newDate();           
+            }
+            else {
+                myDate = new Date(newDate);
+                dateSet = true;                                     
+            }
+            myDate.setHours(0, 0, 0, 0);    
+        },
+        getDayName: function() {            
+            var weekday = ["Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"];
+            if (!dateSet) {
+                this.setDate();
+            }
+            var dayName =  weekday[myDate.getDay()];           
+            return dayName;
+        },
+        getMonthName: function() {          
+            var month = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+            if (!dateSet) {
+                this.setDate();
+            }
+            var monthName =  month[myDate.getMonth()];             
+            return monthName;
+        },
+        isFuture: function() {
+            if (!dateSet) {
+                this.setDate();
+            }
+            var d = new Date();
+            d.setHours(0, 0, 0, 0);
+            return (myDate.valueOf() > d.valueOf());             
+        },
+        isToday: function()  {
+            if (!dateSet) {
+                this.setDate();
+            }
+            var d = new Date();
+            d.setHours(0, 0, 0, 0);
```

Comment:
This line is unnecessary and changes the time when you don't want it to. To figure out if it's "today", compare the month, date, and year on your new Date instance d to myDate's month, date, and year.